smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -48,10 +48,12 @@ contract MixinTransfer is
   function shareKey(
     address _to,
     uint _tokenId,
-    uint _timeShared
+    uint _timeShared,
+    address _keyManager
   ) public
     onlyIfAlive
     onlyKeyOwnerOrApproved(_tokenId)
+    onlyKeyManager(_tokenId)
```

Comment:
I think having multiple modifiers using the same params is a little hard to follow.  Can we merge these into a single modifier?

Reply:
This change is made already, just not part of this PR. (It's in with the changes to the `approve` & `setApprovalForAll` permissions.

Reply:
okay that's fine. maybe I should focus on that one first. i'm not sure having both like this is valid

Reply:
NOw that the way forward with approvals is clear I could add it to this pr... it's dependent on this one anyway and doesn't change much. It should simplify review.

Reply:
Okay that works. I just hesitate to merge to master with something like this. An alternate approach we could consider is PRing small parts into a `sm-dev` branch. Then once the entire feature is working we PR that branch to master for a final sanity check.